### PR TITLE
add section on deploying with env variables and multiple aws profiles

### PR DESCRIPTION
**Pull request by kimberrypi**
## Description

add section on deploying with env variables and multiple aws profiles

### CODE REVIEW

File: docs/docs/deploying-to-s3-cloudfront.md [hunk 1]
```diff
@@ -54,6 +54,23 @@ And finally, add the deployment script to your `package.json`:
 That's it!
 Run `npm run build && npm run deploy` to do a build and have it immediately deployed to S3!
 
+## Deploying with `.env` variables
+
+Some deployments requires passing environment variables. To deploy with environment variables, update the deployment script to your `package.json`:
```
Comment (marcysutton) [inline docs/docs/deploying-to-s3-cloudfront.md]:
```suggestion
Some deployments require passing environment variables. To deploy with environment variables, update the deployment script in your `package.json` file:
```

File: docs/docs/deploying-to-s3-cloudfront.md [hunk 2]
```diff
@@ -54,6 +54,23 @@ And finally, add the deployment script to your `package.json`:
...
+
+```js
+"scripts" : {
+    ...
+    "deploy": "yarn -n \"-r dotenv/config\" && yarn build && gatsby-plugin-s3 deploy"
```
Comment (marcysutton) [inline docs/docs/deploying-to-s3-cloudfront.md]: The rest of the doc uses npm. Can you update it to be more consistent?

File: docs/docs/deploying-to-s3-cloudfront.md [hunk 3]
```diff
@@ -54,6 +54,23 @@ And finally, add the deployment script to your `package.json`:
...
+}
+```
+
+This command requires running dotenv first before deploying to s3. Additionally, if you have multiple AWS profiles in your machine, you can deploy by declaring your `AWS_PROFILE` before the deploy script:
+
+```sh
+AWS_PROFILE=yourprofilename yarn deploy
```
Comment (marcysutton) [inline docs/docs/deploying-to-s3-cloudfront.md]: Same as above -- this should use npm.

File: docs/docs/deploying-to-s3-cloudfront.md [hunk 4]
```diff
@@ -54,6 +54,23 @@ And finally, add the deployment script to your `package.json`:
...
+Some deployments requires passing environment variables. To deploy with environment variables, update the deployment script to your `package.json`:
+
+```js
+"scripts" : {
+    ...
+    "deploy": "yarn -n \"-r dotenv/config\" && yarn build && gatsby-plugin-s3 deploy"
+}
+```
+
+This command requires running dotenv first before deploying to s3. Additionally, if you have multiple AWS profiles in your machine, you can deploy by declaring your `AWS_PROFILE` before the deploy script:
```
Comment (marcysutton) [inline docs/docs/deploying-to-s3-cloudfront.md]:
```suggestion
This command requires running `dotenv` first before deploying to s3. Additionally, if you have multiple AWS profiles in your machine, you can deploy by declaring your `AWS_PROFILE` before the deploy script:
```

**Comment by kimberrypi** [comment]
Hello, for some reason my commit mixed up with the others. I created another PR for this one:
#18045

**Pull request closed without merging**